### DISCUSSION TITLE
Yanking (again) and deleting

### DISCUSSION THREAD

**Discussion by Praczet** [ggandor/leap.nvim#145] [Q&A] [answered]
Hi,

I have another question. When I am leaping forward  let say to the first instance of word 'is' in the text :
`-- This is a dummy text {with a lot of is and this}`
From the first character my motion is: `siss` and it positions the  cursor on the letter `i` so it works as I would expected, but when yanking or deleting is involved it deletes (or yanks) until the letter `s`  (included) so in this case for the motion `dsiss` it will delete text: `-- This is`. 
So my question is: How to setup that it yanks/deletes until the letter `i` (exclusively)?

**Comment by ggandor** [comment]
`vim.keymap.set({'x', 'o'}, 's', function () require('leap').leap { inclusive_op = true } end)` (that is, remove the `offset = 1` argument from `<Plug>(leap-forward-to)`). See `:help leap.leap()`' and `:help leap-custom-mappings`.

**Reply by Praczet** [reply]
Thank you for the replay. 
I have checked it. So in visual mode indeed cursor lands on the first letter. But still for yank and delete it behaves as I described before (deletes until the second character included). I have added del to make sure that this works I mean if I have only `vim.keymap.del` it removes mapping, and `vim.kemap.set` adds it but with behavior described above


```lua
vim.keymap.del({ "x", "o" }, "s")
vim.keymap.set({ "x", "o" }, "s", function()
  require("leap").leap({ inclusive_op = true })
end)
```

I even tried to add `offset = -1` but it changes nothing

Thank you for your patience

**Reply by ggandor** [reply]
Check `:omap s`. Did you add the new mapping after `add_default_mappings()`?

**Reply by Praczet** [reply]
so `:omap s` indicates line with the code that I've added. 
This one. 
```lua
vim.keymap.del({ "x", "o" }, "s")
vim.keymap.set({ "x", "o" }, "s", function()
  require("leap").leap({ inclusive_op = true })
end)
``` 

I will check the leap-forward-till...

**Comment by ggandor** [comment, marked as answer]
> How to setup that it yanks/deletes until the letter i (exclusively)?

Oh, you mean you don't want to include `i`? Then simply use `<Plug>(leap-forward-till)`.

**Reply by Praczet** [reply]
Hi again, 

So my config looks like this

```lua
vim.keymap.del({ "x", "o" }, "s")
vim.keymap.set({ "x", "o" }, "s", "<Plug>(leap-forward-till)")
```

`:omap s` gives 

![Screenshot from 2023-05-03 09-42-11](https://user-images.githubusercontent.com/109667910/235857924-730de213-1da7-4f25-aea7-61f681a76d3b.png)

And with which-keys for the `d` motion I have something like this:


![Screenshot from 2023-05-03 09-40-20](https://user-images.githubusercontent.com/109667910/235858014-e5379f48-b57e-4a36-8ba3-a0949629875a.png)

So it looks like `s` has `<Plug>(leap-forward-till)`

But when I type `ds` nothing happen, it looks like it is going out of deleting mod.

Best regards

**Reply by Praczet** [reply]
It was really annoying me why it doesn't work when it should... So I tried again.. This time I listening my own advice and I used configuration from LazyVim... 
This is what I did, I...
1. added file `leap.lua` in plugins folder `(~/.config/nvim/lua/plugins/)`
2. put configuration from LazyVim
3. Remove mapping for s
4. add mapping as György suggested

My leap.lua file:

```lua
return {
  {
    "ggandor/leap.nvim",
    keys = {
      { "s", mode = { "n", "x", "o" }, desc = "Leap forward to" },
      { "S", mode = { "n", "x", "o" }, desc = "Leap backward to" },
      { "gs", mode = { "n", "x", "o" }, desc = "Leap from windows" },
    },
    config = function(_, opts)
      local leap = require("leap")
      for k, v in pairs(opts) do
        leap.opts[k] = v
      end
      leap.add_default_mappings(true)
      vim.keymap.del({ "x", "o" }, "x")
      vim.keymap.del({ "x", "o" }, "X")
      vim.keymap.del({ "x", "o" }, "s")
      vim.keymap.set({ "x", "o" }, "s", "<Plug>(leap-forward-till)")
    end,
  },
}
```
